File: Leetcode/70.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n <= 0) {
            return 0;
        }
        using ll = long long;
        constexpr int maxn = 90;
        vector <ll> A;
        A.push_back(1);
        A.push_back(2);
        while (A.size() <= maxn) {
            int n = A.size();
            A.push_back(A[n - 1] + A[n - 2]);
        }
        return static_cast<int>(A[n - 1]);
    }
};

File: Leetcode/70/70.go
const N int = 46

func climbStairs(n int) int {
	dp := [N]int{}
	dp[0] = 0
	dp[1] = 1
	dp[2] = 2
	for i := 3; i <= n; i++ {
		dp[i] = dp[i-1] + dp[i-2]
	}
	return dp[n]
}